mlir/lib/Target/LLVMIR/Dialect/OpenMP/OpenMPToLLVMIRTranslation.cpp

```diff
@@ -3836,6 +3895,58 @@ convertOmpTarget(Operation &opInst, llvm::IRBuilderBase &builder,
   bool isOffloadEntry =
       isTargetDevice || !ompBuilder->Config.TargetTriples.empty();
 
+  // For some private variables, the MapsForPrivatizedVariablesPass
+  // creates MapInfoOp instances. Go through the private variables and
+  // the mapped variables so that during codegeneration we are able
+  // to quickly look up the corresponding map variable, if any for each
+  // private variable.
+  if (!targetOp.getPrivateVars().empty() && !targetOp.getMapVars().empty()) {
+    auto argIface = llvm::cast<omp::BlockArgOpenMPOpInterface>(*targetOp);
+    OperandRange privateVars = targetOp.getPrivateVars();
+    std::optional<ArrayAttr> privateSyms = targetOp.getPrivateSyms();
+    std::optional<ArrayAttr> privateMapIdices = targetOp.getPrivateMapsAttr();
+
+    for (auto [privVarIdx, privVarSymPair] :
+         llvm::enumerate(llvm::zip_equal(privateVars, *privateSyms))) {
+      auto privVar = std::get<0>(privVarSymPair);
+      auto privSym = std::get<1>(privVarSymPair);
+
+      SymbolRefAttr privatizerName = llvm::cast<SymbolRefAttr>(privSym);
+      omp::PrivateClauseOp privatizer =
+          findPrivatizer(targetOp, privatizerName);
+
+      if (!privatizer.needsMap())
+        continue;
+
+      assert(privateMapIdices.has_value() &&
+             privateMapIdices->size() == privateVars.size());
+      int64_t mapInfoOpIdx =
+          llvm::cast<IntegerAttr>((*privateMapIdices)[privVarIdx]).getInt();
```

Comment:
I would add a helper function to the MLIR operation. Probably going so far as something like
```
Value getMappedValueForPrivateVar(Value);
```

Or if the O(n) lookup of the variable feels too wasteful, at least something like
```
Value getMappedValueForPrivateVar(unsigned i);
```

Some of the other helpers for private block arguments have helped readability in my opinion e.g. "getCopyMoldArg" and "getDeallocMoldArg".